src/main/resources/db_scripts/liquibase/v0.4.4.refactor-job-parameters.sql

```diff
@@ -0,0 +1,103 @@
+FROM job_template jt
+WHERE jd.job_template_id = jt.id;
+
+
+
+
+
+
+
+alter table job_template alter column variables drop not null;
+alter table job_template alter column variables set default '{}';
+alter table job_template alter column maps drop not null;
+alter table job_template alter column maps set default '{}';
+alter table job_template alter column key_value_pairs drop not null;
+alter table job_template alter column key_value_pairs set default '{}';
+
+alter table job_template add job_parameters jsonb not null default '{}';
+update job_template
+set job_parameters = variables::jsonb || maps::jsonb || key_value_pairs::jsonb;
+
+alter table job_template rename column variables to variables_old;
+alter table job_template rename column maps to maps_old;
+alter table job_template rename column key_value_pairs to key_value_pairs_old;
+
+update job_template
+set job_parameters = job_parameters::jsonb || ('{"jobType":"' || job_type || '"}' )::jsonb;
+
+alter table job_template rename column job_type to job_type_old;
+alter table job_template alter column job_type_old drop not null;
+
+update workflow_history
+set workflow = jsonb_set(workflow::jsonb, array['dagDefinitionJoined', 'jobDefinitions'], updated.updatedJobDefinitions)
+    from (
+  select
+    historyId,
+    jsonb_agg(
+      jsonb_set(
+        toUpdateWithFormConfig.jobDefinition::jsonb,
+        array ['jobParameters'],
+        (toUpdateWithFormConfig.jobDefinition::jsonb -> 'jobParameters' -> 'maps')::jsonb ||
+        (toUpdateWithFormConfig.jobDefinition::jsonb -> 'jobParameters' -> 'variables')::jsonb ||
+        (toUpdateWithFormConfig.jobDefinition::jsonb -> 'jobParameters' -> 'keyValuePairs')::jsonb ||
+        ('{"formConfig":' || toUpdateWithFormConfig.formConfigName || '}')::jsonb,
+        true
+      )
+    ) as updatedJobDefinitions
+  from (
+    select
+      toUpdate.historyId,
+      toUpdate.jobDefinition,
+      case
+        when toUpdate.jobTypeName::varchar is not null
+        then toUpdate.jobTypeName::varchar
+        when jobTemplate.form_config::varchar  is not null
+        then concat('"', jobTemplate.form_config::varchar,'"')
+        else '"Unknown"'::varchar
+      end as formConfigName
+    from (
+      select
+        id as historyId,
+        jsonb_array_elements(workflow::jsonb -> 'dagDefinitionJoined' -> 'jobDefinitions')  as jobDefinition,
+        jsonb_array_elements(workflow::jsonb -> 'dagDefinitionJoined' -> 'jobDefinitions') -> 'jobType' -> 'name' as jobTypeName,
+        jsonb_array_elements(workflow::jsonb -> 'dagDefinitionJoined' -> 'jobDefinitions') -> 'jobTemplateId' as templateId
+      from workflow_history
+    ) as toUpdate
+    left outer join job_template as jobTemplate
+    on toUpdate.templateId::bigint = jobTemplate.id
+  ) as toUpdateWithFormConfig
+  group by toUpdateWithFormConfig.historyId
+) as updated
+where updated.historyId=workflow_history.id;
```

Comment:
Btw, you could make this script idempotent by adding this line
```
  and workflow_history.workflow::jsonb -> 'dagDefinitionJoined' -> 'jobDefinitions' -> 'jobParameters' ? 'maps';
```
It shouldn't be a problem in production, but during testing, I (accidentally) reran the update query twice and then the jobDefinitions of all workflow histories were set to `[null]`

Reply:
Done, but as discussed it has to go deeper into the script because 'jobDefinitions' is an array